File: platform/diff-syntax/src/main/java/de/thomasrosenau/diffplugin/DiffFileType.java
/*
 Copyright 2020 Thomas Rosenau

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */

package de.thomasrosenau.diffplugin;

import javax.swing.*;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class DiffFileType extends LanguageFileType {
    public static final DiffFileType INSTANCE = new DiffFileType();

    private DiffFileType() {
        super(DiffLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Diff/Patch";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Diff/Patch";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "diff";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return DiffIcons.FILE;
    }
}

File: platform/diff-impl/src/com/intellij/diff/editor/DiffFileType.kt
// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
package com.intellij.diff.editor

import com.intellij.icons.AllIcons
import com.intellij.openapi.diff.DiffBundle
import com.intellij.openapi.fileTypes.FileType
import javax.swing.Icon

class DiffFileType private constructor() : FileType {
  override fun getName(): String = "DIFF"
  override fun getDescription(): String = DiffBundle.message("filetype.diff.description")
  override fun getDefaultExtension(): String = ""
  override fun getIcon(): Icon = AllIcons.Actions.Diff

  override fun isBinary(): Boolean = true
  override fun isReadOnly(): Boolean = true

  companion object {
    val INSTANCE: FileType = DiffFileType()
  }
}
